apps/services/auth/delegation-api/src/app/domains/test/test-cases.ts

```diff
@@ -0,0 +1,308 @@
+import {
+  createCurrentUser,
+  createNationalId,
+} from '@island.is/testing/fixtures'
+import { AuthScope } from '@island.is/auth/scopes'
+import { User } from '@island.is/auth-nest-tools'
+import { ConfigType } from '@island.is/nest/config'
+import { DelegationConfig } from '@island.is/auth-api-lib'
+import {
+  CreateCustomDelegation,
+  CreateDomain,
+} from '../../../../test/fixtures/types'
+
+type DomainAssertion = {
+  name: string
+  scopes: Array<{ name: string }>
+}
+
+type TestCase = {
+  user: User
+  customScopeRules?: ConfigType<typeof DelegationConfig>['customScopeRules']
+  delegations?: CreateCustomDelegation[]
+  accessTo?: string[]
+  domains: CreateDomain[]
+  expected: DomainAssertion[]
+}
+
+export const testCases: Record<string, TestCase> = {
+  // Normal user should be able to grant delegations for scopes allowing explicit delegation grants.
+  happyCase: {
+    user: createCurrentUser({ scope: [AuthScope.delegations] }),
+    domains: [
+      {
+        name: 'd1',
+        apiScopes: [
+          {
+            name: 's1',
+            allowExplicitDelegationGrant: true,
+          },
+        ],
+      },
+    ],
+    expected: [
+      {
+        name: 'd1',
+        scopes: [
+          {
+            name: 's1',
+          },
+        ],
+      },
+    ],
+  },
+  // Should not see scopes unless they allow explicit delegation grants.
+  noExplicitDelegationGrant: {
+    user: createCurrentUser({ scope: [AuthScope.delegations] }),
+    domains: [
+      {
+        name: 'd1',
+        apiScopes: [
+          {
+            name: 's1',
+            allowExplicitDelegationGrant: false,
+          },
+        ],
+      },
+    ],
+    expected: [],
+  },
+  // Procuring holder should only see scopes they have access to.
+  procuringHolderScopes: {
+    user: createCurrentUser({
+      nationalIdType: 'company',
+      delegationType: 'ProcurationHolder',
+      scope: [AuthScope.delegations],
+    }),
+    accessTo: ['s1'],
```

Comment:
Is this really testing access controlled scopes?

Reply:
No, this is most likely a copy paste issue. Great catch!